torch/csrc/jit/serialization/pickler.cpp

```diff
@@ -352,6 +352,40 @@ void Pickler::pushTensor(const IValue& ivalue) {
   }
 }
 
+void Pickler::pushLiteralSparseTensor(const at::Tensor& tensor) {
+  // The arguments to this function are:
+  // size, requires_grad, pinned memory, data_type, device,
+  // indices, values, backward_hooks
+  pushGlobal("torch._utils", "_rebuild_sparse_coo_tensor");
+  // size
+  push<PickleOpCode>(PickleOpCode::MARK);
+  push<PickleOpCode>(PickleOpCode::MARK);
+  for (auto size : tensor.sizes()) {
+    pushInt(size);
+  }
+  push<PickleOpCode>(PickleOpCode::TUPLE);
```

Comment:
curious, why do we need two `PickleOpCode::TUPLE`?

Reply:
My understanding is that `PickleOpCode::TUPLE` is used after a `PickleOpCode::MARK` to create an immutable list.  In the `pushLiteralSparseTensor` since there are two `PickleOpCode::MARK` two `PickleOpCode::TUPLE` are required.

Reply:
The first `PickleOpCode::MARK` is used for the sparse tensor. The second `PickleOpCode::MARK` is used for `tensor.sizes()`, the definition is `at::IntArrayRef sizes()`.